src/storage/storage.cc

```diff
@@ -51,7 +51,13 @@ class StorageImpl : public Storage {
   static void ActivateDevice(Context ctx) {
```

Comment:
This method should not be static. It is not protected by any RAII object. Besides, it is not used by any other classes anyway.